kong/include/kong/services/negotiation/v1/negotiation.proto

```diff
@@ -0,0 +1,46 @@
+syntax = "proto3";
+
+option go_package = "github.com/kong/koko/internal/gen/wrpc/kong/model;model";
+
+package kong.model;
+
+
+// NegotiationService allows DP to ask for services and CP answer with the accepted version of each
+// +wrpc:service-id=5
+service NegotiationService {
+  // NegotiateServices should be the first RPC call upon connecting.  It allows the CP
+  // to get which services does the DP handle.
+  //
+  // Call direction: DP to CP
+  // +wrpc:rpc-id=4
+  rpc NegotiateServices(kong.model.NegotiateServicesRequest) returns (kong.model.NegotiateServicesResponse);
+}
+
+message NegotiateServicesRequest {
+  DPNodeDescription node = 1;
+  map<string, ServiceVersions> services_requested = 2;
+}
+
+message DPNodeDescription {
+  string type = 2;
+  string version = 3;
+  string hostname = 4;
+}
+
+// When version is undefined, service is denied.
+message ServiceVersion {
+  string version = 1;
+  string description = 2;
+}
```

Comment:
at least put a boolean instead of overloading `version` with two different meanings.

still strongly prefer two separate arrays, since the fields actually mean different things, and could easily evolve differently.

Reply:
I think it's intuitive: when there's no version(it being nil), then there's not a suitable version to use.

Reply:
I don't have so strong opinion on whether to introduce a boolean for that, but after consideration, I still insist to use one list instead of two. The meaning of the list is obvious: the answers of CP for all services.

Reply:
due to differences between Lua and protobuf, there's too much ambiguity between nil and empty strings.  even booleans can be somewhat deceiving (but easier to fix).  overloading a string field will fail as soon as some other module in Lua changes the configuration of the protobuf module (yes, the configs are global).

two arrays is the most robust solution

Reply:
2 solutions without too much thinking:
1. Since `""` is not a good version name, we treat `""` in the same way as `nil` and consider that as not supported(and it doesn't matter what configuration `pb` is in);
2. Or just a boolean to tell if service is accepted;

I think the 2 array solution is too verbose, but if it has a strong reason I'm OK with that solution.

Reply:
I think Javier has given a good reason for this issue. and I think this needs not too much change.